subsys/net/ip/tcp.c

```diff
@@ -349,6 +353,28 @@ static void tcp_send(struct net_pkt *pkt)
 	tcp_pkt_unref(pkt);
 }
 
+static void tcp_derive_rto(struct tcp *conn)
+{
+#ifdef CONFIG_NET_TCP_RANDOMIZED_RTO
+	/* Compute a randomized rto 1 and 1.5 times tcp_rto */
+	uint32_t gain;
+	uint8_t gain8;
+	uint32_t rto;
+
+	/* Getting random is computational expensive, so only use 8 bits */
+	sys_rand_get(&gain8, sizeof(uint8_t));
+
+	gain = (uint32_t)gain8;
+	gain += 1 << 9;
+
+	rto = (uint32_t)tcp_rto;
+	rto = (gain * rto) >> 9;
```

Comment:
@ssharks I'm not familiar with net/tcp stuff but you are doing here a double assigning. Are you sure it is intentional?

Shouldn't it be?
<pre>
rto = (uint32_t)tcp_rto;
rto += (gain * rto) >> 9;
</pre>

Reply:
No problem, I started to doubt as well, but the +1 is already incorporated in the gain as you might have spotted. The other way could have also worked indeed, but than you have to omit line 368 as well.